benchmark/kvbench/runtime/etcd_rt.py

```
@@ -209,29 +259,45 @@ def all_reduce(
+                    if time.time() - start_time > timeout_sec:
+                        raise TimeoutError(
+                            f"[Rank {self.rank}] all_reduce timeout waiting for rank {dest_rank} after {timeout_sec}s"
+                        )
+                    time.sleep(0.05)
+                    val = self.client.get(f"{self.prefix}/all_reduce/{dest_rank}")[0]
+                all_vals.append(pickle.loads(val))
+
+            logger.debug("All reduce values: %s", all_vals)
             if op == ReduceOp.SUM:
-                final_val = [sum(col) for col in zip(*vals)]
+                final_val = [sum(col) for col in zip(*all_vals)]
```

Ruff (0.15.13)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)